screens/ChatScreen.js

```diff
@@ -32,12 +33,12 @@ class ChatScreen extends Component {
   };
 
   async componentDidMount() {
-    const token = await AsyncStorage.getItem('hm_token');
-    this.state.id = `${JSON.parse(token)._id}`;
+    const { hmToken } = this.props.user;
+    this.state.id = `${hmToken._id}`;
 
     // Init Twilio service and message service
     this.newMessagesCallback = this.newMessagesCallback.bind(this);
-    this.twilioService = new TwilioService(token, [this.state.contact], this.newMessagesCallback);
+    this.twilioService = new TwilioService(hmToken, [this.state.contact], this.newMessagesCallback);
```

Comment:
It looks like this is throwing an error in its current form.

SyntaxError: JSON Parse error: Unexpected identifier "object"]

Stack trace:
  services/twilioService.js:11:28 in TwilioService
  screens/ChatScreen.js:41:43 in componentDidMount$

Reply:
It looks like it's a token parsing issue that we probably want to adjust. Most of the ChatScreen logic will get altered when we update the backend, but the token parsing logic on the client side will be the same.

Reply:
this happens because we need to pass a string, I stringified the JSON object